subprojects/core/src/test/groovy/org/gradle/api/internal/file/collections/ImmutableFileCollectionTest.groovy

```diff
@@ -32,10 +32,12 @@ import java.util.concurrent.Callable
 @UsesNativeServices
 class ImmutableFileCollectionTest extends Specification {
     def 'can create empty collection'() {
-        ImmutableFileCollection files = ImmutableFileCollection.of()
+        ImmutableFileCollection collection1 = ImmutableFileCollection.of()
+        ImmutableFileCollection collection2 = ImmutableFileCollection.of(new File[0])
 
         expect:
-        files.files.size() == 0
+        collection1.files.size() == 0
+        collection2.files.size() == 0
```

Comment:
You could even test that `collection1 is collection2`

Reply:
that's what the next test method does :)